src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -499,23 +493,32 @@ internal ImmutableArray<IOperation> CreateIgnoredDimensions(BoundNode declaratio
             {
                 case BoundKind.LocalDeclaration:
                     {
-                        return CreateFromArray<BoundExpression, IOperation>(((BoundLocalDeclaration)declaration).DeclaredTypeOpt.BoundDimensionsOpt);
+                        BoundTypeExpression? declaredTypeOpt = ((BoundLocalDeclaration)declaration).DeclaredTypeOpt;
```

Comment:
It's a shame that we can't express in the type system that this will never be null.

If backcompat wasn't an issue we could presumably declare a `new` non-nullable version of the property on the derived type to express that? I wonder if this is going to be a common issue with nullable going forward where we have no way to express not null for a member of a derived type.

Reply:
Hmm, actually could we not use `[MemberNotNull]` on the ctor of the `BoundLocalDeclaration`? I assume it could be null in error scenarios, so perhaps when `hasErrors == false`?

Reply:
It's not about backcompat (as this is a BoundNode, which we can change at will).

Reply:
Right, so should we alter the bound nodes then to express the correct nullability?

Reply:
Feel free to file a followup bug :)